rewrite-core/src/main/java/org/openrewrite/text/CreateTextFile.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.text;
+
+import org.openrewrite.*;
+import org.openrewrite.marker.Markers;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+@Incubating(since = "7.12.0")
+public class CreateTextFile extends Recipe {
+
+    @Option(displayName = "File Contents",
+            description = "Multiline text content for the file.",
+            example = "Some text.")
+    private final String fileContents;
+
+    @Option(displayName = "Relative File Path",
+            description = "File path of new file.",
+            example = "foo/bar/baz.txt")
+    private final String relativeFileName;
+
+    @Option(displayName = "Overwrite existing file",
+            description = "If there is an existing file, should it be overwritten.")
+    private final Boolean overwriteExisting;
```

Comment:
Let's make this `required = false`, add a `@Nullable` annotation on it. Later on where we do the check, do it as `Boolean.FALSE.equals(overwriteExisting)`. I think in every other case, the default nullable value is `true`, so not sure if you want to invert the language here or not. I think it could go either way.

Reply:
I kept the language the same to avoid what felt like trying to reason with double negatives. Added an explicit `null` check early in the method as a part of the bail-early condition. Doing nothing if there was not express consent to overwrite felt like the _do no harm_ approach.